### DISCUSSION TITLE
When Device Swap occured?

### DISCUSSION THREAD

**Discussion by bigludo7** [camaraproject/DeviceSwap#11] [General]
Hello team
(cc: @jgarciahospital @krishvenkatachalam @rartych @StefanoFalsetto-CKHIOD)

We need probably to have alignement about when we consider that a device swap occurred. Looking for your perspective.

Here a first proposal to be discussed:

**When Device Swap event is triggered:**
- (Nominal case) An active SIM (with previous traffic) is removed from device A to be introduced to Device B - Device B is turn on.
- A sim swap is performed (phone number stays same) and the new sim is introduced in a distinct device that was previous sim  (note: we have here both a device swap and a sim swap).
 
 
**When Device Swap event is not triggered**
- A sim swap is performed (phone number stays same) and the new sim is introduced in same device that was previous sim.
- A change of phone number for an active sim (with previous traffic) is performed without change of device.
- A change of phone number & sim card (with previous traffic) is performed without change of device.
 
The case of a new 'acquisition"  (new line with new sim & new device) need to be discussed - do we considered it as a device swap, a 'special' device swap, or not a device swap....

**Comment by StefanoFalsetto-CKHIOD** [comment]
Hi Ludovic,
maybe we can quickly find an answer by generalizing the definition of “device swap” using the link between the MSISDN (entry point of the API) and IMEI (unique identifier of the device).
In my opinion, the “device swap” can be generally defined as:
The date of the last MSISDN/IMEI pairing change.
In that way we can correctly include any case of MSISDN change, independently from the SIM, but taking in consideration the pairing with a specific device. Ideally, in case a device swap never happened, this attribute will contain the date of the first time the Subscriber’s SIM has been put into his/her mobile phone.

Moreover, I think the word “swap” of this API name could be misleading. It implies the use of two devices, but we know that’s not the case.

––
Stefano Falsetto
Digital Service Architect



Da: Ludovic Robert ***@***.***>
Data: venerdì, 23 agosto 2024 alle ore 09:55
A: camaraproject/DeviceSwap ***@***.***>
Cc: Stefano Falsetto, Con ***@***.***>, Mention ***@***.***>
Oggetto: [camaraproject/DeviceSwap] When Device Swap occured? (Discussion #11)

Hello team
(cc: @jgarciahospital<https://github.com/jgarciahospital> @krishvenkatachalam<https://github.com/krishvenkatachalam> @rartych<https://github.com/rartych> @StefanoFalsetto-CKHIOD<https://github.com/StefanoFalsetto-CKHIOD>)

We need probably to have alignement about when we consider that a device swap occurred. Looking for your perspective.

Here a first proposal to be discussed:

When Device Swap event is triggered:
·         (Nominal case) An active SIM (with previous traffic) is removed from device A to be introduced to Device B - Device B is turn on.
·         A sim swap is performed (phone number stays same) and the new sim is introduced in a distinct device that was previous sim (note: we have here both a device swap and a sim swap).

When Device Swap event is not triggered
·         A sim swap is performed (phone number stays same) and the new sim is introduced in same device that was previous sim.
·         A change of phone number for an active sim (with previous traffic) is performed without change of device.
·         A change of phone number & sim card (with previous traffic) is performed without change of device.

The case of a new 'acquisition" (new line with new sim & new device) need to be discussed - do we considered it as a device swap, a 'special' device swap, or not a device swap....

—
Reply to this email directly, view it on GitHub<https://github.com/camaraproject/DeviceSwap/discussions/11>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BDPZ62I3W7VNEI2H7HI7YETZS3TFTAVCNFSM6AAAAABM7U4HRCVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGA4DKNZTGE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Please consider the environment before printing this email
________________________________
This e-mail is only intended for the person(s) to whom it is addressed and may contain PRIVILEGED or CONFIDENTIAL information. Any opinions or views are personal to the writer and do not represent those of CKH IOD UK Limited, CK Hutchison Holdings Limited or its group companies. If you are not the intended recipient, you are hereby notified that any use, retention, disclosure, copying, printing, forwarding or dissemination of this communication is strictly prohibited. If you have received this communication in error, please erase all copies of the message and its attachments and notify the sender immediately. CKH IOD UK Limited is a company registered in England and Wales with company number 11324057. Registered Office: Hutchison House, 5 Hester Road, Battersea, London. SW11 4AN. www.ckhiod.com<http://www.ckhiod.com/>

**Reply by bigludo7** [reply]
Thanks @StefanoFalsetto-CKHIOD 
Yes this is simplest to define and simpler is often better.

One case probably borderline but perhaps can exist: Change phoneNumber for an existing 'line'(without changing device) - following your proposal we have a Device Swap where technically this is not really one. But perhaps this case is marginal.

**Reply by bigludo7** [reply]
I look on mobile connect specification and saw the device-change is when Timestamp of last MSISDN <-> IMEI pairing change

So aligned with your view @StefanoFalsetto-CKHIOD  - works for me to use this rule.

**Reply by jgarciahospital** [reply]
Totally aligned with this proposal, any scenario where MSISDN-IMEI pairing change (leaving apart multisim by the moment).

So, previous scenarios and maybe some others could look like this:

1. A recently created SIM&line (MSISDN) is installed in a device (new or not)-> Device Swap, new installation also modifies the binding of MSISDN and IMEI (from none association to one).
2. A SIM&line(MSISDN) that was initially installed in device-A is switched to device-B -> Device Swap
3. A new SIM&line (MSISDN-B) is installed in a device where the SIM&line (MSISDN-A) was originally installed -> Device Swap for MSISDN B, but no for MSISDN A until it's connected in a new device.
4. A new SIM is binded to an existing line (MSISDN suffers a SIM Swap) and it's installed in the same device -> NO Device Swap, only SIM Swap (as IMSI change, but not IMEI change)
5. A new SIM is binded to an existing line (MSISDN suffers a SIM Swap) and it's installed in a new device -> Device Swap (and also SIM Swap, as change in IMEI and IMSI binding)
6. A SIM&line(MSISDN-A) installed in a device suffers a modification, in which that SIM gets associated to MSISDN-B -> Similar to case 3, Device Swap in MSISDN-B as new installation (like in case 1), but not for MSISDN-A which will only report a Device Swap after it's installed in other device

**Reply by bigludo7** [reply]
@jgarciahospital I went thru the scenario and look good.... at this end of the day this is always the same rule - any new [MSISDN,IMEI] pair triggers a Device swap.

**Comment by jgarciahospital** [comment]
Proposal to include in API description, avoiding explicit Multi-SIM additional discussion (only including it as new device IMEI is associated to same MSISDN). We can later follow the discussion on considering only new multiSIM devices or consider any change.

**Device swap**: A Device Swap is a process in which a user's mobile phone number (MSISDN) is associated with a new device (IMEI).

This typically occurs when the user switches to a new mobile device, either by upgrading, replacing a lost or damaged device, or other similar reasons. A Device Swap also happens during actions like activating a new device associated with the same phone number, also known as multiSIM. New Device activation is considered as a Device Swap as well, as the phone number is associated with a device it was not associated before.
